### PR TITLE
libretro update

### PR DESCRIPTION
**Pull request by natinusala**
Notable changes:
- stella renamed to stella2014
- new libs for easyrpg and desmume
- easyrpg now uses cmake
- the system overlay folder is always created in RetroArch to avoid failing the unit if PPSSPP is disabled (cc @mlnlbrt )

Not all cores will build for all platforms, I'm waiting for the PR bot to tell me where it fails.

### CODE REVIEW

File: packages/libretro/libretro-database/package.mk [hunk 1]
```diff
@@ -19,7 +19,7 @@
 ################################################################################
 
 PKG_NAME="libretro-database"
-PKG_VERSION="e23ab29"
+PKG_VERSION="5809b27"
```
Comment (RobLoach) [inline packages/libretro/libretro-database/package.mk]: A few database updates...

``` suggestion
PKG_VERSION="2a8e537"
```

Comment (natinusala) [reply packages/libretro/libretro-database/package.mk]: I did run the update script a week ago, thanks for the heads up

**Comment by ToKe79** [comment]
Hi,
this PR (e758286c3ce8f403e33aa8b69db6521b1ab5244a) did not build correctly for any target - all (45) builds failed.
Logs can be found here: http://nightly.builds.lakka.tv/PRs/pr_764_e758286/

_This is an automated message_

**Comment by ToKe79** [comment]
Hi,
this PR (0609fa378aaba13dd42279a5af3e7000282e4a7a) did not build correctly for all targets.
Following targets failed (43):
- Generic.i386
- Generic.x86_64
- Generic_VK_nvidia.x86_64
- RPi.arm
- RPi2.arm
- Allwinner.Bananapi.arm
- Allwinner.Cubieboard2.arm
- Allwinner.Cubietruck.arm
- Allwinner.orangepi_2.arm
- Allwinner.orangepi_lite.arm
- Allwinner.orangepi_one.arm
- Allwinner.orangepi_pc.arm
- Allwinner.orangepi_plus.arm
- Allwinner.orangepi_plus2e.arm
- Allwinner.nanopi_m1_plus.arm
- imx6.cuboxi.arm
- imx6.udoo.arm
- OdroidC1.arm
- Odroid_C2.arm
- WeTek_Core.arm
- WeTek_Hub.arm
- WeTek_Play.arm
- WeTek_Play_2.arm
- Gamegirl.arm
- S8X2.S82.arm
- S8X2.M8.arm
- S8X2.T8.arm
- S8X2.MXIII-1G.arm
- S8X2.MXIII-PLUS.arm
- S8X2.X8H-PLUS.arm
- S805.MXQ.arm
- S805.HD18Q.arm
- S805.M201C.arm
- S805.M201D.arm
- S805.MK808B-Plus.arm
- S905.arm
- S912.arm
- Rockchip.TinkerBoard.arm
- Rockchip.ROCK64.arm
- Rockchip.ROC-RK3328-CC.arm
- Rockchip.MiQi.arm
- Rockchip.ROCKPro64.arm
- Rockchip.ROCK960.arm

Logs and images for testing can be found here: http://nightly.builds.lakka.tv/PRs/pr_764_0609fa3/

_This is an automated message_

**Comment by Ghabry** [comment]
For EasyRPG Player you are using now CMake with system libraries. This is awesome and the recommended way to do it (instead of the ugly stuff in builds/libretro).
Because you are using CMake now you can actually use the upstream repository under https://github.com/easyrpg/player (and pin to tag 0.6.1), it contains all the code necessary for a CMake libretro build with system libraries.

Though you are missing all the optional dependencies which means audio playback will not work. When looking at the logfile:

``
-- Could NOT find Harfbuzz (missing: HARFBUZZ_LIBRARY HARFBUZZ_INCLUDE_DIR) 
-- Could NOT find speexdsp (missing: SPEEXDSP_LIBRARY SPEEXDSP_INCLUDE_DIR) 
-- Could NOT find Samplerate (missing: SAMPLERATE_LIBRARY SAMPLERATE_INCLUDE_DIR) 
-- Could NOT find mpg123 (missing: MPG123_LIBRARY MPG123_INCLUDE_DIR) 
-- Could NOT find LibSndFile (missing: LIBSNDFILE_LIBRARY LIBSNDFILE_INCLUDE_DIR) 
-- Could NOT find Vorbisfile (missing: VORBISFILE_LIBRARY VORBISFILE_INCLUDE_DIR) 
-- Could NOT find Opusfile (missing: OPUSFILE_LIBRARY OPUSFILE_INCLUDE_DIR) 
-- Could NOT find WildMidi (missing: WILDMIDI_LIBRARY WILDMIDI_INCLUDE_DIR) 
-- Could NOT find XMP (missing: XMP_LIBRARY XMP_INCLUDE_DIR) 
-- 
-- Target system: libretro
-- 
-- Audio backend: libretro
-- 
-- WAV playback: None
-- MIDI playback: built-in (FmMidi)
-- MP3 playback: None
-- Ogg Vorbis playback: None
-- MOD playback: None
-- Opus playback: None
-- Resampler: None
-- 
-- Font rendering: Freetype and built-in
-- 
-- Configuring done
``

**Comment by natinusala** [comment]
@Ghabry thanks for the heads up, I will add as many as I can.

We try to keep Lakka as minimal as possible, could you point me to the most important libs among all the missing ones? If we can avoid adding all of them that could be great

**Comment by ToKe79** [comment]
Hi,
this PR (76ec343b9b5a106842a742fd174c764c5f18fd6e) did not build correctly for all targets.
Following targets failed (1):
- Generic.i386

Logs and images for testing can be found here: http://nightly.builds.lakka.tv/PRs/pr_764_76ec343/

_This is an automated message_

**Comment by Ghabry** [comment]
@natinusala 

Okay. The problem is that RPG Maker 2000 games can throw any junk at you, so we need libraries that support all kinds of audio formats.

What is necessary for most games:
- **libspeexdsp** (Audio resampling, otherwise the playback speed will be wrong for WAV files)
- **mpg123** (MP3 playback, most games need this)
- **libsndfile** (WAV playback), if you don't want this dependency you can (WARNING: Not recommended! There is a reason this is not exposed as an option in CMake) set the CXXFLAG "-DWANT_FASTWAV", this enables the built-in WAV decoder which is very limited and doesn't support 24bit PCM or ADPCM, so many files will fail to play

Nice to have:
- **harfbuzz** (Will be needed to play games with "complex scripts" like Thai, but everything else will work without it)
- **vorbisfile** (Not many games use OGG because this format is not officially supported by RPG Maker, but there are unofficial patches which added it, which is the reason why there are a few games with this)
- **opusfile** (No game uses this, because there is no OPUS support by official RPG Maker)
- **wildmidi** (We have a built-in MIDI decoder, so MIDI will work without it but users will complain that it sounds strange and it is very CPU hungry. Btw the MIDI GUS patches stuff is available via RetroArch content downloader so you don't need to bundle them)
- **libxmp(-lite)** (Not many games use MOD/Tracker Audio files because it is not officially supported by RPG Maker, but there are unofficial patches which added it)

**Comment by natinusala** [comment]
Thanks a lot for the detailed answer!

harfbuzz needs cairo so I'm going to skip this one.

I have that in the logs: `MIDI playback: WildMidi. Fallback to built-in (FmMidi)` 

That means that WildMidi has been recognized and it being used right?

As for libxmp, the repo doesn't have any configure script nor autogen.sh, not sure how I would go about building it...

**Comment by Ghabry** [comment]
> harfbuzz needs cairo

When you compile it manually this dependency can be disabled, but as I said harfbuzz is not highly critical.

> I have in the logs
Yes, it will use WildMidi and when for some reason these GUS Instrument files are missing it uses FmMidi automatically.

> libxmp
https://github.com/cmatsuoka/libxmp

You "autoreconf -fi" to generate a configure script.

Alternatively, libxmp is kinda annoying to compile, you can use libxmp-lite from http://sourceforge.net/projects/xmp/files/libxmp/4.4.1/libxmp-lite-4.4.1.tar.gz

If you have issues with the configure script you can use this CMake file instead (maintained by us): https://github.com/EasyRPG/buildscripts/blob/master/shared/CMakeLists_xmplite.txt

**Comment by natinusala** [comment]
Okay it was super easy with the CMake script. However building easyrpg now fails:

```
easyrpg_libretro.so: error: undefined reference to 'xm_loader'
easyrpg_libretro.so: error: undefined reference to 'it_loader'
easyrpg_libretro.so: error: undefined reference to 's3m_loade
```

**Comment by Ghabry** [comment]
my bad, you also need this patch: https://github.com/EasyRPG/buildscripts/blob/master/shared/libxmp-a0288352.patch  :)

**Comment by natinusala** [comment]
It's almost fixed, now I only have

`easyrpg_libretro.so: error: undefined reference to 'xmp_set_player'`

**Comment by Ghabry** [comment]
@natinusala 
This seems to be caused by ``#define USE_VERSIONED_SYMBOLS`` in src/common.h, line 14. When the define is removed it should work.

No idea though why you have this problem, we also compile everything from source for various platforms and never encountered this linker error :/.

**Comment by natinusala** [comment]
It works, thanks :)

**Comment by Ghabry** [comment]
Looks good to me :)

**Comment by ToKe79** [comment]
I will run the job again, as due to the non-existence of the GLideN64 branch on the remote for mupen64plus_next (failure during git pull - see fix in #770) the job failed on most all targets (except those, wchich do not pack mupen64plus_next, or were built after I have fixed the repo folder manually).

**Comment by ToKe79** [comment]
Hi,
this PR (353c5f602686e02df62ebef705dfa755c672a7f3) did not build correctly for all targets.
Following targets failed (43):
- Generic.i386
- Generic.x86_64
- Generic_VK_nvidia.x86_64
- RPi2.arm
- RPi2.RPi4.arm
- Allwinner.Bananapi.arm
- Allwinner.Cubieboard2.arm
- Allwinner.Cubietruck.arm
- Allwinner.orangepi_2.arm
- Allwinner.orangepi_lite.arm
- Allwinner.orangepi_one.arm
- Allwinner.orangepi_pc.arm
- Allwinner.orangepi_plus.arm
- Allwinner.orangepi_plus2e.arm
- Allwinner.nanopi_m1_plus.arm
- imx6.cuboxi.arm
- imx6.udoo.arm
- OdroidC1.arm
- Odroid_C2.arm
- OdroidXU3.arm
- WeTek_Core.arm
- WeTek_Hub.arm
- WeTek_Play.arm
- WeTek_Play_2.arm
- S8X2.S82.arm
- S8X2.M8.arm
- S8X2.T8.arm
- S8X2.MXIII-1G.arm
- S8X2.MXIII-PLUS.arm
- S8X2.X8H-PLUS.arm
- S805.MXQ.arm
- S805.HD18Q.arm
- S805.M201C.arm
- S805.M201D.arm
- S805.MK808B-Plus.arm
- S905.arm
- S912.arm
- Rockchip.TinkerBoard.arm
- Rockchip.ROCK64.arm
- Rockchip.ROC-RK3328-CC.arm
- Rockchip.MiQi.arm
- Rockchip.ROCKPro64.arm
- Rockchip.ROCK960.arm

Logs and images for testing can be found here: http://nightly.builds.lakka.tv/PRs/pr_764_353c5f6/

_This is an automated message_

**Comment by natinusala** [comment]
The Mupen64 source folder needs to be cleaned again it seems

**Comment by ToKe79** [comment]
Hi,
this PR (353c5f602686e02df62ebef705dfa755c672a7f3) did not build correctly for all targets.
Following targets failed (43):
- Generic.i386
- Generic.x86_64
- Generic_VK_nvidia.x86_64
- RPi2.arm
- RPi2.RPi4.arm
- Allwinner.Bananapi.arm
- Allwinner.Cubieboard2.arm
- Allwinner.Cubietruck.arm
- Allwinner.orangepi_2.arm
- Allwinner.orangepi_lite.arm
- Allwinner.orangepi_one.arm
- Allwinner.orangepi_pc.arm
- Allwinner.orangepi_plus.arm
- Allwinner.orangepi_plus2e.arm
- Allwinner.nanopi_m1_plus.arm
- imx6.cuboxi.arm
- imx6.udoo.arm
- OdroidC1.arm
- Odroid_C2.arm
- OdroidXU3.arm
- WeTek_Core.arm
- WeTek_Hub.arm
- WeTek_Play.arm
- WeTek_Play_2.arm
- S8X2.S82.arm
- S8X2.M8.arm
- S8X2.T8.arm
- S8X2.MXIII-1G.arm
- S8X2.MXIII-PLUS.arm
- S8X2.X8H-PLUS.arm
- S805.MXQ.arm
- S805.HD18Q.arm
- S805.M201C.arm
- S805.M201D.arm
- S805.MK808B-Plus.arm
- S905.arm
- S912.arm
- Rockchip.TinkerBoard.arm
- Rockchip.ROCK64.arm
- Rockchip.ROC-RK3328-CC.arm
- Rockchip.MiQi.arm
- Rockchip.ROCKPro64.arm
- Rockchip.ROCK960.arm

Logs and images for testing can be found here: http://nightly.builds.lakka.tv/PRs/pr_764_353c5f6/

_This is an automated message_

**Comment by ToKe79** [comment]
Hi,
this PR (704047136ceed49e7b039066efa49f4478a83658) did not build correctly for all targets.
Following targets failed (1):
- Generic.i386

Logs and images for testing can be found here: http://nightly.builds.lakka.tv/PRs/pr_764_000068/

_This is an automated message_

**Comment by ToKe79** [comment]
Hi,
this PR (c7c642cb5d2f2876bc4b68e2506e5a638f21cca5) did not build correctly for all targets.
Following targets failed (2):
- Generic.i386
- Generic.x86_64

Logs and images for testing can be found here: http://nightly.builds.lakka.tv/PRs/pr_764_000070/

_This is an automated message_

**Comment by natinusala** [comment]
I fixed retroarch (easyrpg now requires sdl2 and it was picking it up, enabling unwanted sdl stuff)

I also added flycast_wince to all platforms (will eventually replace flycast) and bsnes_hd, also to all platforms

**Comment by Ghabry** [comment]
EasyRPG Player shouldn't pick up SDL2, that's just for the standalone version (the one without libretro API)

**Comment by natinusala** [comment]
It's one of the easyrpg dependencies that have sdl2 as a dependency

**Comment by Ghabry** [comment]
ah right. Forgot about indirect dependencies due to libraries from the repository. nvm

**Comment by ToKe79** [comment]
Hi,
this PR (fb4ebd66c68192ad1d7f11bb40fa7bf89f2e8fc7) did not build correctly for all targets.
Following targets failed (4):
- Generic_VK_nvidia.x86_64
- RPi.arm
- RPi.GPICase.arm
- Gamegirl.arm

Logs and images for testing can be found here: http://nightly.builds.lakka.tv/PRs/pr_764_000072/

_This is an automated message_

**Comment by ToKe79** [comment]
Hi,
this PR (b45336061ef474499950592be6c3d5f433bd5f25) did not build correctly for all targets.
Following targets failed (4):
- Generic_VK_nvidia.x86_64
- RPi.arm
- RPi.GPICase.arm
- Gamegirl.arm

Logs and images for testing can be found here: http://nightly.builds.lakka.tv/PRs/pr_764_000073/

_This is an automated message_

**Comment by natinusala** [comment]
@ToKe79 it looks like the last build is a fluke (I successfully build RPi locally), could you restart it please?

**Comment by ToKe79** [comment]
Hi,
this PR (0b6064417f1531c7e58d50cba7b0967b56a495cb) did not build correctly for any target - all (46) builds failed.
Logs can be found here: http://nightly.builds.lakka.tv/PRs/pr_764_000077/

_This is an automated message_

**Pull request merged**